packages/features/insights/server/routing-events.ts

```diff
@@ -0,0 +1,459 @@
+                  },
+                }),
+          }
+        : {}),
+      ...(fieldFilter && {
+        response: {
+          path: [fieldFilter.fieldId, "value"],
+          array_contains: [fieldFilter.optionId],
+        },
+      }),
+      form: formsTeamWhereCondition,
+    };
+
+    const totalResponsePromise = prisma.app_RoutingForms_FormResponse.count({
+      where: responsesWhereCondition,
+    });
+
+    const responsesPromise = prisma.app_RoutingForms_FormResponse.findMany({
+      select: {
+        id: true,
+        response: true,
+        form: {
+          select: {
+            id: true,
+            name: true,
+          },
+        },
+        routedToBooking: {
+          select: {
+            uid: true,
+            status: true,
+            createdAt: true,
+            attendees: {
+              select: {
+                timeZone: true,
+                email: true,
+              },
+            },
+            user: {
+              select: { id: true, name: true, email: true, avatarUrl: true },
+            },
+          },
+        },
+        createdAt: true,
+      },
+      where: responsesWhereCondition,
+      orderBy: {
+        createdAt: "desc",
+      },
+      take: limit ? limit + 1 : undefined, // Get one extra item to check if there are more pages
+      cursor: cursor ? { id: cursor } : undefined,
+    });
+
+    const [totalResponses, responses] = await Promise.all([totalResponsePromise, responsesPromise]);
+
+    // Parse response data
+    const parsedResponses = responses.map((r) => {
+      const responseData = routingFormResponseInDbSchema.parse(r.response);
+      return { ...r, response: responseData };
+    });
+
+    return {
+      total: totalResponses,
+      data: parsedResponses,
+      nextCursor: responses.length > (limit ?? 0) ? responses[responses.length - 1].id : undefined,
+    };
+  }
+
+  static async getRoutingFormFieldOptions({
+    teamId,
+    isAll,
+    routingFormId,
+    organizationId,
+  }: RoutingFormInsightsTeamFilter) {
+    const formsWhereCondition = await this.getWhereForTeamOrAllTeams({
+      teamId,
+      isAll,
+      routingFormId,
+      organizationId,
+    });
+
+    const routingForms = await prisma.app_RoutingForms_Form.findMany({
+      where: formsWhereCondition,
+      select: {
+        id: true,
+        fields: true,
+      },
+    });
+
+    const fields = routingFormFieldsSchema.parse(routingForms.map((f) => f.fields).flat());
+
+    return fields;
+  }
+
+  static async getFailedBookingsByRoutingFormGroup({
+    teamId,
+    isAll,
+    routingFormId,
+    organizationId,
+  }: RoutingFormInsightsTeamFilter) {
+    const formsWhereCondition = await this.getWhereForTeamOrAllTeams({
+      teamId,
+      isAll,
+      organizationId,
+      routingFormId,
+    });
+
+    const teamConditions = [];
+
+    // @ts-expect-error it doest exist but TS isnt smart enough when its unmber or int filter
```

Comment:
This is where it gets interesting - constructing raw SQL for teams conditions based on the above logic